auction/sort.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package auction
+
+// SortUint64 implements sorting by uint64 keys for
+// canonical encoding of maps in msgpack format.
+//msgp:ignore SortUint64
+//msgp:sort uint64 SortUint64
+type SortUint64 []uint64
```

Comment:
What's the purpose of this type?

Reply:
As in, I don't think it shows up anywhere in the change. Is this type required?

Reply:
Also, a more general question--will `msgp` complain if it tries to serialize map with keys whose type I have not defined a sort function for? (Does this apply to all non `msgp:ignore`d types in the package?)

Reply:
The `SortUint64` type implements `sort.Interface` for sorting `uint64`s.  The `msgp:sort` annotation in the comment above tells the msgp code-generator to use this type to sort map keys of type `uint64`.

I don't believe `msgp` will complain, but the generated go code for serializing a map will not compile if the key type (or the `//msgp:sort` annotation for the key type) doesn't implement `sort.Interface`.

Reply:
Oh, and the `//msgp:ignore` annotation tells msgp not to generate encoders/decoders for that type.  This seems appropriate here because I don't expect any encoded/decoded data structures to contain `SortUint64`.

Reply:
To clarify, it seems that there are three `sort.go` files (inside of `auction/`, `crypto/` and `data/basics/`), which specify sort interfaces for `uint64`, `Digest`, and `AssetIndex`, respectively. `git grep SortAssetIndex` shows up, as expected,
```
derek@crassus:~/newgo/go-algorand$ git grep SortAssetIndex
data/basics/msgp_gen.go:                        sort.Sort(SortAssetIndex(zb0001_keys))
data/basics/msgp_gen.go:                        sort.Sort(SortAssetIndex(zb0003_keys))
data/basics/msgp_gen.go:                        sort.Sort(SortAssetIndex(zb0001_keys))
data/basics/msgp_gen.go:                        sort.Sort(SortAssetIndex(zb0003_keys))
...
```
but the other two do not show up in generated code:
```
derek@crassus:~/newgo/go-algorand$ git grep SortDigest | grep -v auction/sort.go | grep -v crypto/sort.go | wc -l
0
derek@crassus:~/newgo/go-algorand$ git grep SortUint64 | grep -v auction/sort.go | grep -v crypto/sort.go | wc -l
0
```

Reply:
Ah, I see.  Good catch!  I think they are remnants of data structures that used to be there but went away.

In particular, `SortDigest` was originally used because `auction.RunningAuction` has a `map[crypto.Digest]BidderState`.  But then I added support for `//msgp:ignore` to avoid codegen for that type altogether, since it's never actually encoded.  And I never went back to clean up the `SortDigest` thing.

The `SortUint64` used to be required when we had assets named by `uin64`, but then the `AssetIndex` type name got introduced for it, and I didn't clean up the previous `SortUint64` code.

I'll remove both of those unused sorts.  Thanks!